### ISSUE TITLE
sc-tagger - cannot retrieve client_id

### ISSUE THREAD

**Issue by t3mulligan** [Sliim/soundcloud-syncer#5]
The solution provided in the other issue, works for sc-syncer, however it doesn't appear there is an argument for client_id or user on sc-tagger. (my mp3's aren't getting tagged when running sc-syncer.)

**Comment by Sliim** [comment]
I pushed a fix for `sc-tagger`, can you try if it fix your issue please?
For `sc-syncer`, look like there is a regression caused by a change in soundcloud's api response, I will push another fix for that.

**Comment by Sliim** [comment]
I pushed the fix for `sc-syncer`. Let me know if its ok for you ;) And thanks again for reporting that!

**Comment by t3mulligan** [comment]
Ok, started fresh, cloned the repo. Ran the following: `sc-syncer -u 100001 -c 2000000000000000z -o mp3 -p`

My account has a single playlist of 77 songs. Here's the output: `=========================
Processing playlist test!
=========================


Downloading Steve Aoki & Louis Tomlinson - Just Hold On (Two Friends Remix) (311660083)..
100% 3686400 / 3685563
Downloaded => mp3/steveaoki/311660083-just-hold-on-two-friends-remix-1.mp3`
There is one of those for each song. no other output whatsoever. Identical output as before. No tags.

sc-tagger command: `scripts/sc-tagger -c 200000000000000z mp3folder`

output: `Process tag for `Captain Cuts - Love Like We Used To (Young Bombs Remix)` :id => 310379069
Done. 45 processed, 0 in error.`

There are 45 of the 'process' output lines, nothing else I didn't include. Tagging worked, sweet! Thanks!
The playlist(77 songs) downloads 45 mp3's 1 m4a 30 wav's and 1 aiff. All are recent uploads (past 6 months or so). Only the mp3's get tagged. Going to try with --convert again, last time it failed partway through, but I hadn't yet realized the different audio formats, so never checked which format it failed on, will report back on that along with output (might take a bit to convert.) Thanks again!

**Comment by t3mulligan** [comment]
Ok, same ss-syncer command as before, but with --convert. I forgot there was one song with a .raw extension(forgot because I just renamed it to wav.) Here's the error output: 
```
Downloading Echosmith - Cool Kids (Gazzo & Two Friends Remix) (190293922)..
100% 3997696 / 3991927
Downloaded => mp3/two-friends/190293922-echosmith-cool-kids-gazzo-two-friends-remix-2.raw
Converting to mp3/two-friends/190293922-echosmith-cool-kids-gazzo-two-friends-remix-2.mp3..
Traceback (most recent call last):
  File "scripts/sc-syncer", line 140, in <module>
    main()
  File "scripts/sc-syncer", line 137, in main
    args.limit)
  File "scripts/sc-syncer", line 38, in downloader
    downloader(splaylist.get_tracks, output, process_tag, convert_file, max_retry, *args)
  File "scripts/sc-syncer", line 47, in downloader
    strack.convert()
  File "/usr/local/lib/python3.4/dist-packages/ssyncer/strack.py", line 251, in convert
    song = AudioSegment.from_file(backupfile)
  File "/usr/local/lib/python3.4/dist-packages/pydub/audio_segment.py", line 452, in from_file
    sample_width = kwargs['sample_width']
KeyError: 'sample_width'
```

Aiff, and m4a and wav worked just fine; however is there a way to avoid re-encoding the mp3's? Thanks.

**Comment by Sliim** [comment]
Do you have `soundcloud-syncer` installed from `pip`? If yes you should upgrade the package with this command inside the repo: `pip install --upgrade .`. This will upgrade the `ssync/strack` lib where I pushed the fix, and you should be able to tag mp3 files with `sc-syncer`.

> Only the mp3's get tagged
Yes only mp3 are supported for ID3 tags.

For the convert issue, I need to investigate what happen with `pydub`. Give me some time.

**Comment by t3mulligan** [comment]
I haven't used pip no. Also in regards to the 'raw' files, I'm not positive but I'm pretty sure all of the soundcloud raw's can just be renamed to .wav. Regardless of the exact reason for the issue, as I've always understood it wav is just a raw uncompressed audio file, and renaming worked just fine for me manually; so that might be the simplest solution.

**Comment by Sliim** [comment]
I can't reproduce with raw extensions, here is my output for same track you get the convert error:
```
Downloading Echosmith - Cool Kids (Gazzo & Two Friends Remix) (190293922)..
100% 3997696 / 3991927
Downloaded => /data/tmp/test-sc/two-friends/190293922-echosmith-cool-kids-gazzo-two-friends-remix-2.mp3
Processing tags for /data/tmp/test-sc/two-friends/190293922-echosmith-cool-kids-gazzo-two-friends-remix-2.mp3..
```
I suspect a bug here: https://github.com/Sliim/soundcloud-syncer/blob/master/ssyncer/strack.py#L107

Can you give me your python version with `python --version` and the `magic` python package version `pip freeze | grep magic` please? Thank you.

**Comment by Sliim** [comment]
I tried with python `3.4.0`, same output above.
```
$ python --version                                                                          
Python 3.4.0
$ pip freeze | grep magic
python-magic==0.4.12
```

**Comment by Sliim** [comment]
I improved the method to get file extension, normally, you should download the track id `190293922` as a `mp3` file, not `raw`. Can you retry with latest `master` changes please? Thx

**Comment by t3mulligan** [comment]
Ok, I didn't see anywhere in this repository about downloading a specific track. Is there a way to do this, or did you just mean to do likes or playlist containing that track? I will be free all afternoon starting in an hour so will look into it further, however, I do have console output for you. (different track)
Here is the command I'm running: (5th track in this playlist)
`soundcloud-syncer# scripts/sc-syncer -u 139260146 -c 2xxxxxxxxxxxxz -o mp3 -p -O 7 --convert`
Here is the output using the most recent master:
```
Downloading SHIROSKY - Fallin' (191393541)..
100% 3866624 / 3860270
Downloaded => mp3/shirosky_official/191393541-shirosky-fallin.raw
Converting to mp3/shirosky_official/191393541-shirosky-fallin.mp3..
Traceback (most recent call last):
  File "scripts/sc-syncer", line 140, in <module>
    main()
  File "scripts/sc-syncer", line 137, in main
    args.limit)
  File "scripts/sc-syncer", line 38, in downloader
    downloader(splaylist.get_tracks, output, process_tag, convert_file, max_retry, *args)
  File "scripts/sc-syncer", line 47, in downloader
    strack.convert()
  File "/usr/local/lib/python3.4/dist-packages/ssyncer/strack.py", line 251, in convert
    song = AudioSegment.from_file(backupfile)
  File "/usr/local/lib/python3.4/dist-packages/pydub/audio_segment.py", line 452, in from_file
    sample_width = kwargs['sample_width']
KeyError: 'sample_width'
```

**Comment by Sliim** [comment]
> Ok, I didn't see anywhere in this repository about downloading a specific track. Is there a way to do this, or did you just mean to do likes or playlist containing that track?

Yes this is what I mean. Sorry.

I'm not sure you upgraded correctly `soundcloud-syncer` :s 
Can you explain me how you installed `soundcloud-syncer`? Or are you just pulling the repository and run sc-syncer by: `scripts/sc-syncer ..`? In this case ensure there is no `*.pyc` files in the `ssyncer/` directory.

**Comment by Sliim** [comment]
With latest `master` changes, the output should look like:
```
> sc-syncer -u Sliim -o /tmp/test-sc -L 1 --client-id xxx --convert

Downloading SHIROSKY - Fallin' (191393541)..
100% 3866624 / 3860270
Downloaded => /tmp/test-sc/shirosky_official/191393541-shirosky-fallin.mp3
File is already in mp3 format. Skipping convert.
Processing tags for /tmp/test-sc/shirosky_official/191393541-shirosky-fallin.mp3..
```

**Comment by t3mulligan** [comment]
Ah ok, my apologies. I have been cloning the repo fresh each time, creating a new directory for mp3, and running the command from soundcloud-syncer/

I attempted to install it via pip, and had no luck, and have been cloning the repo manually since. I thought I had removed the pip installation, but it turns out pip was no longer symlinked to pip3, pip uninstall soundcloud-syncer found no such installation to remove, however I just realized pip3 uninstall did actually uninstall the package. Sorry for the confusion, I have no python/pip experience at all.

After uninstalling via pip(3) I'm getting an error when running from cloned repository script, here is the output:
```
Traceback (most recent call last):
  File "scripts/sc-syncer", line 20, in <module>
    from ssyncer.sclient import sclient
ImportError: No module named 'ssyncer'
```
I'm sure I can figure it out, no need to waste your time on it, will post my solution here, unless you already know my problem. Thanks!

**Comment by Sliim** [comment]
No problem! You're welcome ;)

Try with:
```
PYTHONPATH=ssync/ scripts/sc-syncer ..
```
You get this error because `ssync` lib is missing from your python path (since uninstall), this means that you was using the old lib from the pip installation :).

You can also install current sources from your local repository with `pip3 install .` (`pip3 install --upgrade .` if already installed)

**Comment by Sliim** [comment]
I released version `0.3.4` to pypi: https://pypi.python.org/pypi/soundcloud-syncer/0.3.4
This release includes `sc-tagger` fixes.
